checkout-core/src/main/java/com/adyen/checkout/core/DispatcherProvider.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 16/10/2024.
+ */
+
+package com.adyen.checkout.core
+
+import androidx.annotation.RestrictTo
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+object DispatcherProvider {
```

Comment:
Nice, this concept is probably the best we can do right now! 

I think we could improve it a little bit by making these `var`s lazy. This will make sure the dispatchers are not initialised on the app's startup, possibly increasing startup times. Something like this:
```Kotlin
private var mainFactory = { Dispatchers.Main }
val Main: MainCoroutineDispatcher get() = mainFactory()

private var defaultFactory = { Dispatchers.Default }
val Default: CoroutineDispatcher get() = defaultFactory()

private var ioFactory = { Dispatchers.IO }
val IO: CoroutineDispatcher get() = ioFactory()

fun overrideMain(dispatcher: MainCoroutineDispatcher) {
    mainFactory = { dispatcher }
}

fun resetMain() {
    mainFactory = { Dispatchers.Main }
}

...
```

Reply:
I like it. done!